File: src/cryptotax/api/schemas/parse.py
from decimal import Decimal

from pydantic import BaseModel


class ParseTestRequest(BaseModel):
    tx_hash: str


class ParsedSplitResponse(BaseModel):
    account_label: str
    account_type: str
    symbol: str
    quantity: Decimal


class ParseTestResponse(BaseModel):
    tx_hash: str
    parser_name: str
    entry_type: str
    splits: list[ParsedSplitResponse]
    balanced: bool
    warnings: list[str]


class ParseWalletResponse(BaseModel):
    processed: int
    errors: int
    total: int


class ParseStatsResponse(BaseModel):
    total: int
    parsed: int
    errors: int
    unknown: int

File: web/src/api/parse.ts
import { apiFetch, withEntityId } from './client'

export interface ParsedSplit {
  account_label: string
  account_type: string
  symbol: string
  quantity: number
}

export interface ParseTestResponse {
  tx_hash: string
  parser_name: string
  entry_type: string
  splits: ParsedSplit[]
  balanced: boolean
  warnings: string[]
}

export interface ParseWalletResponse {
  processed: number
  errors: number
  total: number
}

export interface ParseStatsResponse {
  total: number
  parsed: number
  errors: number
  unknown: number
}

export async function parseTest(txHash: string, entityId?: string): Promise<ParseTestResponse> {
  return apiFetch<ParseTestResponse>(withEntityId('/parse/test', entityId), {
    method: 'POST',
    body: JSON.stringify({ tx_hash: txHash }),
  })
}

export async function parseWallet(walletId: string, entityId?: string): Promise<ParseWalletResponse> {
  return apiFetch<ParseWalletResponse>(withEntityId(`/parse/wallet/${walletId}`, entityId), {
    method: 'POST',
  })
}

export async function getParseStats(entityId?: string): Promise<ParseStatsResponse> {
  return apiFetch<ParseStatsResponse>(withEntityId('/parse/stats', entityId))
}
